packages/app/src/app/hooks/useSubscription.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  SubscriptionStatus,
+  SubscriptionType,
+  TeamMemberAuthorization,
+} from 'app/graphql/types';
+import { useAppState } from 'app/overmind';
+
+export const useSubscription = () => {
+  const {
+    activeTeam,
+    activeTeamInfo,
+    personalWorkspaceId,
+    activeWorkspaceAuthorization,
+  } = useAppState();
+
+  /**
+   * Personal states
+   */
+  const isPersonalSpace = activeTeam === personalWorkspaceId;
+
+  /**
+   * Team states
+   */
+  const isTeamSpace = !isPersonalSpace;
+
+  const isTeamAdmin =
+    isTeamSpace &&
+    activeWorkspaceAuthorization === TeamMemberAuthorization.Admin;
+
+  /**
+   * Subscription states
+   */
+
+  // There are different statuses for a subscription, but only ACTIVE and TRIALING
+  // should be considered an active TeamPro subscription.
+  // TODO: This might change based on how we use other statuses in the subscription (eg: PAUSED)
+  const hasActiveSubscription =
+    activeTeamInfo?.subscription?.status === SubscriptionStatus.Active ||
+    activeTeamInfo?.subscription?.status === SubscriptionStatus.Trialing;
+
+  const hasPastSubscription = Boolean(activeTeamInfo?.subscription?.status);
```

Comment:
would a better name for this be: `hasExistingOrPastSubscription`?

Reply:
Yeah we can definitely do that!